use-selected-items-hook/src/useSelectedItems.ts

```diff
@@ -1,34 +1,53 @@
 import { useState, useCallback, useEffect } from "react";
 import update from "immutability-helper";
 
-import { HookArguments, HookReturnValues } from "./types";
+import { HookArguments, HookReturnValues, Item } from "./types";
 
 function useSelectedItems<T>({
   items,
   itemIdentifier,
 }: HookArguments<T>): HookReturnValues<T> {
   const [selectedItems, setSelectedItems] = useState<T[]>([]);
+  const [itemsList, setItemsList] = useState<Item<T>[]>([]);
 
-  const [itemsList, setItemsList] = useState(() => (
-    items.map(item => ({
-      ...item,
-      selected: false,
-    }))
-  ));
-
-  const verifyIfIdentifierIsValid = useCallback(() => {
-    const identifierIsValid = !!(items.find(findItem => (
+  useEffect(() => {
+    const itemIdentifierIsInvalid = !(items?.find(findItem => (
       findItem[itemIdentifier]
     )));
 
-    if (!identifierIsValid) {
+    const hasItems = !!items?.length;
+
+    if (itemIdentifierIsInvalid && hasItems) {
       throw new Error("Please, make sure to provide a valid identifier");
```

Comment:
Probably better to use `invariant` instead of `if (...) { throw new Error(...) }`.

Check it out here: https://www.npmjs.com/package/invariant

Also, note that the condition is inverted (if `false`, it throws).

So:

```jsx
if (itemIdentifierIsInvalid && hasItems) {
      throw new Error("Please, make sure to provide a valid identifier");
}
```

Becomes (adding `hasItems` to `itemIdentifierIsInvalid` eval):

Note: please validate expression below!

```jsx
const isValid = !items?.length || (
  !!(items?.find(findItem => (
    findItem[itemIdentifier]
  )))
);

invariant(isValid, "Please, make sure to provide a valid identifier");
```

Reply:
Should I add some babel plugin in order to minify these errors in production environment?

Reply:
Applied in 4b71eed6e9a1d00ddb28f886d0f0b09a922c674e